src/main/java/com/quizapp/enterprise/services/QuizService.java

```diff
@@ -20,11 +21,13 @@
 
 
     @Override
-    public Quiz createQuiz(Quiz quizToAdd, String username) {
+    public Quiz createQuiz(Quiz quizToAdd, String username) throws Exception {
 
-       int userId =  userRepository.findUserByEmail(username).getUserID();
-
-       quizToAdd.setUserId(userId);
+       User user = userRepository.findUserByEmail(username);
+       if(user == null){
+           throw new Exception("user not found");
+       }
+       quizToAdd.setUserId(user.getUserID());
```

Comment:
included a check for a possible null prior to execution on possible null object so code may handle the case as desired.
This change was recommended by the "8. Proper handling of Null Pointer Exceptions"  section of  the java coding best practices summary referenced below

source:https://xperti.io/blogs/java-coding-best-practices/